tekton/ci/shared/go_coverage.yaml

```diff
@@ -0,0 +1,215 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: go-coverage
+  annotations:
+    description: |
+      Runs go coverage of a git repo using the knative coverage tool.
+spec:
+  params:
+    - name: pullRequestNumber
+      description: The number of the pull request to check out.
+    - name: gitRevision
+      description: sha of the commit making the pull request
+    - name: jobType
+      description: Type of prow job - presubmit/postsubmit/periodic.
+      default: presubmit
+    - name: buildUUID
+      description: Prow type build ID.
+    - name: repoName
+      description: name of the repository
+    - name: repoOwner
+      description: owner of the repo (user/orgaization).
+    - name: jobName
+      description: name of the job.
+    - name: postSubmitGcsBucket
+      description: Name of the gcs bucket where to fetch post submit job from.
+    - name: postSubmitJobName
+      description: Name of the post submit job to compare the coverage against.
+    - name: covTarget
+      description: the relative directory from where to run the coverage.
+      default: "."
+    - name: profileName
+      description: Name of the coverage profile file.
+      default: "coverage_profile.txt"
+    - name: covThresholdPercentage
+      description: coverage percentage below which to mark the coverage test as failed.
+      default: 0
+  workspaces:
+  - name: source
+    description: Location where the git repo will be installed.
+    mountPath: /go/src/github.com/tektoncd
+  - name: github-token
+    description: |
+      A secret workspace where the github personal access token resides.
+      The token should be storage as a stringData with key "token" and
+      value "your access token".
+  steps:
+    - name: clone-repo
+      image: alpine/git
+      script: |
+        #!/bin/sh
+        git clone https://github.com/$(params.repoOwner)/$(params.repoName).git
+        cd $(params.repoName)
+        git fetch origin pull/$(params.pullRequestNumber)/head:$(params.gitRevision)
+        git checkout $(params.gitRevision)
```

Comment:
Every other CI job uses:

```yaml
    - name: clone-repo
      taskRef:
        name: git-batch-merge
        bundle: gcr.io/tekton-releases/catalog/upstream/git-batch-merge:0.2
```

Any reason for not using it here and having a custom cloning logic?

Reply:
I just wasn't aware of it. I updated it to use git-batch-merge instead.